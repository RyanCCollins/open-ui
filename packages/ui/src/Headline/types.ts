import * as React from 'react';
import { Margin } from '../Paragraph/types';
import Headline from './index';
export type TextAligment = 'center' | 'left' | 'right' | 'justify';
export type FontWeight = 100 | 200 | 300 | 400 | 500 | 600 | 700 | 800;
export type HeadlineSize = 'small' | 'medium' | 'xlage' | 'xlarge';
export interface SizeMap {
  small: number;
  medium: number;
  large: number;
  xlarge: number;
};

export interface HeadlineProps extends React.HTMLProps<Headline> {
  color?: string;
  textAlign?: string;
  fontSize?: HeadlineSize;
  fontWeight?: FontWeight;
  margin?: Margin;
}
