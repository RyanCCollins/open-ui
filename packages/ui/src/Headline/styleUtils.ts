import { HeadlineSize, SizeMap } from './types';
import remStringFromPX from '../utils';

const sizeMap: SizeMap = {
  small: 30,
  medium: 48,
  large: 60,
  xlarge: 72,
};

export default function calculateSize(size: HeadlineSize): string {
  return remStringFromPX(sizeMap[size]);
};
