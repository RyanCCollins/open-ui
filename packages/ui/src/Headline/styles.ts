import { marginCss } from '../Paragraph/styles';
import calculateSize from './styleUtils';
const styled = require('styled-components').default;

export const HeadlineStyled = styled.h1`
  text-align: center;
  font-size: ${(props) => calculateSize(props.fontSize)};
  text-align: ${(props) => props.textAlign};
  color: ${(props) => props.color};
  font-weight: ${(props) => props.fontWeight};
  ${(props) => marginCss(props.margin)};
`;
