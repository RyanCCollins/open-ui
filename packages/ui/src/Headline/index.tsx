import * as React from 'react';
import { HeadlineStyled } from './styles';
import { HeadlineProps } from './types';

class Headline extends React.Component<HeadlineProps, undefined> {
  public static defaultProps: HeadlineProps = {
    color: '#000000',
    textAlign: 'center',
    fontSize: 'medium',
    fontWeight: 400,
    margin: 'medium',
  };
  public render() {
    const { children, ...rest } = this.props;
    return (
      <HeadlineStyled {...rest}>
        {children}
      </HeadlineStyled>
    );
  }
}

export default Headline;
