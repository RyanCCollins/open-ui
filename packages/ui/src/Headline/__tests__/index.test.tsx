import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import * as React from 'react';
import Headline from '../';
import props from './__mocks__/headlineProps.mock';

describe('<Headline />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <Headline>
        Headline
      </Headline>,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('should render with different props', () => {
    const wrapper = shallow(
      <Headline {...props}>
        Headline
      </Headline>,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
