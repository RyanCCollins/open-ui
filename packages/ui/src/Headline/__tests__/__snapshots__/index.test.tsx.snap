// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Headline /> should render with default props 1`] = `
<styled.h1
  color="#000000"
  fontWeight={400}
  margin="medium"
  size="medium"
  textAlign="center"
>
  Headline
</styled.h1>
`;

exports[`<Headline /> should render with different props 1`] = `
<styled.h1
  color="#fff"
  fontWeight={400}
  margin="medium"
  size="medium"
  textAlign="center"
>
  Headline
</styled.h1>
`;
